test/unit/queue/src/QueueTest.cpp

```diff
@@ -93,32 +93,45 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(
     Fixture f;
 
     bool CallbackFinished = false;
-    alpaka::queue::enqueue(f.m_queue, [&CallbackFinished]() noexcept {std::this_thread::sleep_for(std::chrono::milliseconds(100u)); CallbackFinished = true;});
+    alpaka::queue::enqueue(
+        f.m_queue,
+        [&CallbackFinished]() noexcept
+        {
+            std::this_thread::sleep_for(std::chrono::milliseconds(100u));
+            CallbackFinished = true;
+        });
 
     alpaka::wait::wait(f.m_queue);
-    BOOST_REQUIRE_EQUAL(true, CallbackFinished);
+    BOOST_CHECK_EQUAL(true, CallbackFinished);
 }
 
 //-----------------------------------------------------------------------------
 BOOST_AUTO_TEST_CASE_TEMPLATE(
-    queueShouldBeEmptyAfterProcessingFinished,
+    queueShouldNotBeEmptyWhenLastTaskIsStillExecutingAndIsEmptyAfterProcessingFinished,
     TDevQueue,
     alpaka::test::queue::TestQueues)
 {
     using Fixture = alpaka::test::queue::QueueTestFixture<TDevQueue>;
     Fixture f;
 
     bool CallbackFinished = false;
-    alpaka::queue::enqueue(f.m_queue, [&CallbackFinished]() noexcept {std::this_thread::sleep_for(std::chrono::milliseconds(100u)); CallbackFinished = true;});
+    alpaka::queue::enqueue(
+        f.m_queue,
+        [&f, &CallbackFinished]() noexcept
+        {
+            BOOST_CHECK_EQUAL(false, alpaka::queue::empty(f.m_queue));
+            std::this_thread::sleep_for(std::chrono::milliseconds(100u));
+            CallbackFinished = true;
+        });
 
     // A synchronous queue will always stay empty because the task has been executed immediately.
     if(!alpaka::test::queue::IsSyncQueue<typename Fixture::Queue>::value)
     {
         alpaka::wait::wait(f.m_queue);
     }
 
-    BOOST_REQUIRE_EQUAL(true, alpaka::queue::empty(f.m_queue));
-    BOOST_REQUIRE_EQUAL(true, CallbackFinished);
+    BOOST_CHECK_EQUAL(true, alpaka::queue::empty(f.m_queue));
```

Comment:
I would have to comment out this line to make the test successful again. I will think about substituting one bug by another one.